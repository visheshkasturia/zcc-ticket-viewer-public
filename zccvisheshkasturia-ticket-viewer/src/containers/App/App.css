.App {
  text-align: center;
  margin: auto;
}

.logout-link{
  text-decoration: none;
  color: black;
  border: solid 1px black;
  border-radius: 8px;
  padding: 10px;
  background-color: #e7e7e7;
  position: fixed;
  right: 20px;
  top: 20px;
}