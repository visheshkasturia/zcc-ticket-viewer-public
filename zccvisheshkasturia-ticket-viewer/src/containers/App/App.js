import React, { useState, useEffect } from 'react';
import './App.css';
import Cookies from 'js-cookie';
import { Navigate } from 'react-router-dom';
import Tickets from '../Tickets/Tickets';

function App() {
  const [redirect, setRedirect] = useState(false);
  const [name, setName] = useState('');
  // fetch cookie if cookie is undefined send to NotFound
  useEffect(() => {
    const cookieData = Cookies.get('zccCookie');
    if (cookieData === undefined) {
      setRedirect(true);
    } else {
      setName(JSON.parse(cookieData.slice(2)).fullName);
    }
  }, []);

  return (
    <div className="App">
      {redirect && <Navigate to="/NotFound" />}
      <Tickets name={name} />
      <a data-testid="logout-link-test" className="logout-link" href="http://localhost:5000/logout">Log out</a>
    </div>
  );
}

export default App;
