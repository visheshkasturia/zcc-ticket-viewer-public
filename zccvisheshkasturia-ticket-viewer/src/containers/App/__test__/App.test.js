import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App renders correctly', () => {
  it('Matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
