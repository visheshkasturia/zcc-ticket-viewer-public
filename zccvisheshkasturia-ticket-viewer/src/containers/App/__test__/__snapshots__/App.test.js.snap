// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly Matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="tickets"
  >
    <h1
      className="userHeading"
    >
       Welcome 
      
       
      <br />
       Thanks for making people happy, one customer at a time.
    </h1>
    <div
      className="pg-info"
    >
      Page 
      1
      , Showing 
      0
        on this page, Total 
      0
       Tickets
    </div>
    <div
      className="ticketContainer"
    >
      <div
        className="id-box flex-center"
      >
        Ticket ID
      </div>
      <div
        className="status-box flex-center"
      >
        Status
      </div>
      <div
        className="priority-box flex-center"
      >
        Priority
      </div>
      <div
        className="subject-box flex-center"
      >
        Subject
      </div>
      <div
        className="date-box flex-center"
      >
        Date
      </div>
    </div>
    <button
      className="nav-btn"
      id="prev-btn"
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
    <button
      className="nav-btn"
      id="next-btn"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
  <a
    className="logout-link"
    data-testid="logout-link-test"
    href="http://localhost:5000/logout"
  >
    Log out
  </a>
</div>
`;
